Doc/c-api/object.rst

```diff
@@ -60,13 +62,46 @@ Object Protocol
    value on success, or ``NULL`` on failure.  This is the equivalent of the Python
    expression ``o.attr_name``.
 
+   If the missing attribute should not be treated as a failure, you can use
+   :c:func:`PyObject_GetOptionalAttr` instead.
+
 
 .. c:function:: PyObject* PyObject_GetAttrString(PyObject *o, const char *attr_name)
 
    Retrieve an attribute named *attr_name* from object *o*. Returns the attribute
    value on success, or ``NULL`` on failure. This is the equivalent of the Python
    expression ``o.attr_name``.
 
+   If the missing attribute should not be treated as a failure, you can use
+   :c:func:`PyObject_GetOptionalAttrString` instead.
+
+
+.. c:function:: int PyObject_GetOptionalAttr(PyObject *obj, PyObject *attr_name, PyObject **result);
+
+   Variant of :c:func:`PyObject_GetAttr` which doesn't raise
+   :exc:`AttributeError`.
+
+   Return ``1`` and set ``*result != NULL`` if the attribute is found.
+   Return ``0`` and set ``*result == NULL`` if the attribute is not found;
+   the :exc:`AttributeError` is silenced.
+   Return ``-1`` and set ``*result == NULL`` if an error other than
+   :exc:`AttributeError` is raised.
```

Comment:
I suggest:

```suggestion
   If the attribute is found, return ``1`` and set *\*result* to a new :term:`strong reference` to the attribute.
   If the attribute is not found, return ``0`` and set *\*result* to ``NULL`` ;
   the :exc:`AttributeError` is silenced.
   If an error other than :exc:`AttributeError` is raised, return ``-1`` and set *\*result* to ``NULL`` .
```

You may not define ``*result`` in the error code path **in the documentation**, it may be left as an implementation detail (and yeah, I also prefer to set it to NULL).